jwala-services/src/main/java/com/cerner/jwala/service/webserver/component/WebServerStateSetterWorker.java

```diff
@@ -113,37 +122,18 @@ public void pingWebServer(final WebServer webServer) {
     }
 
     /**
-     * Checks if the web server is in webServerToPingSet, if it's not in the set it will be added to the set.
-     * This method is thread safe.
-     * @param webServer the web server
-     * @return true if the web server is in webServerToPingSet
-     */
-    private boolean isInWebServerToPingSet(final WebServer webServer) {
-        synchronized (webServersToPing) {
-            if (webServersToPing.contains(webServer.getId())) {
-                LOGGER.debug("List of web servers currently being pinged: {}", webServersToPing);
-                LOGGER.debug("Cannot ping web server {} since it is currently being pinged", webServer.getName(), webServer);
-                return false;
-            }
-            webServersToPing.add(webServer.getId());
-            LOGGER.debug(">>> Added web server {} to a set of web servers that will be pinged", webServer.getName());
-            return true;
-        }
-    }
-
-    /**
      * Checks if web server can be pinged and logs the reason if it cannot be pinged
      * @param webServer the web server
      * @return true if it can be pinged
      */
     private boolean webServerCanBePinged(final WebServer webServer) {
-        if (isWebServerBusy(webServer)) {
-            LOGGER.debug("Cannot ping web server {} since it is busy. Details: {}", webServer.getName(), webServer);
+        if (WebServerReachableState.WS_NEW.equals(webServer.getState())) {
+            LOGGER.debug("Cannot ping web server {} with a status of \"NEW\"", webServer.getName());
             return false;
         }
 
-        if (inMemoryStateManagerService.get(webServer.getId()) == WebServerReachableState.WS_NEW) {
-            LOGGER.debug("Cannot ping web server {} with a status of \"NEW\"", webServer.getName(), webServer);
+        if (isWebServerBusy(webServer)) {
```

Comment:
Can we rename this to isWebServerStartingOrStopping() or something similar since that's what is being checked?
When I see isWebServerBusy() I think it's processing something critical like a client request and shouldn't be pinged.